feat: new policy view with login CTA

kopecs opened the pull request:
The policy view is blank on logging in. This still version of this commit needs some imports updated. Planning on going through and reorganizing imports generally / improving autoformatting and then rebasing probably.

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

| Logged out | Logged in |
|------------|-----------|
|<img width="268" alt="image" src="https://github.com/user-attachments/assets/663f7781-3231-4821-b338-50f42e1dfd06">|<img width="272" alt="image" src="https://github.com/user-attachments/assets/29c261e4-8866-4d0f-b3bb-27c86255839e">|

kopecs commented on src/lsp.ts:
```diff
@@ -249,8 +253,24 @@ async function start(env: Environment): Promise<void> {
     env.emitRulesRefreshedEvent();
   };
 
+  // TODO(cooper): This is an ugly hack to allow for easily updating the login
+  // status.  It is rather fragile and we should instead do this by cleaning up
+  // the LSP extensions we use for login. However, this lets us ensure we have
+  // reasonably up-to-date UI state until then.
```
To be updated per our discussion but I would like to merge this in the interim.

ajbt200128 commented:
screenshots?

ajbt200128 commented on src/views/policy.ts:
```diff
@@ -0,0 +1,21 @@
+    return element;
+  }
+
+  getChildren(
```
what's the source for this data?

kopecs replied:
Oh whoops; sorry there should be placeholder text here in this version. I have more in a future (wip) PR.

kopecs commented:
I'm going to move this back to draft while I update the login flow stuff we discussed since it is causing some cases where the UI would be stale here for now.

kopecs committed:
feat: new policy view with login/add config CTA

This will show a prompt to login or add a local configuration if none is
present. If a configuration is present and/or the user is logged in, it
will show the applied configuration.

kopecs commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/semgrep/semgrep-vscode/186?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#199** <a href="https://app.graphite.dev/github/pr/semgrep/semgrep-vscode/199?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#186** <a href="https://app.graphite.dev/github/pr/semgrep/semgrep-vscode/186?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/semgrep/semgrep-vscode/186?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#187** <a href="https://app.graphite.dev/github/pr/semgrep/semgrep-vscode/187?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

kopecs commented:
### Merge activity

* **Feb 6, 2:40 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/semgrep/semgrep-vscode/186).
